Syntax pitfall leading to fatal warning

Subject: Syntax pitfall leading to fatal warning

### Problem
- I wrote the following in a Sphinx source file:

```
.. figure:: foo.png
    :scale: 80 %
   
   Main window for user ``robin``.
```
but there is a space too much in front of the ```:scale: 80 %```, so reStructuredText does not recognize this as an option and adds it to the filename. This then leads to the following warning:
```/foo/index.rst:18: WARNING: image file not readable: foo.png:scale:80%```

That would'nt be too much of a problem, but it comes worse. I use the ``-W`` option to ask Sphinx to turn warnings into exceptions.
With the ``-W`` option I don't get above warning but the following:

```
Exception occurred:
  File "/py27/local/lib/python2.7/site-packages/sphinx/util/logging.py", line 375, in filter
    message = record.msg % record.args
ValueError: incomplete format
The full traceback has been saved in /tmp/sphinx-err-FITtqd.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile:20: recipe for target 'html' failed
make: *** [html] Error 2
```
This is utterly disturbing because you have no indication about the source location.
The reason has obviously to do with the fact that there is a "%" sign in the error message, which causes logging to expect arguments to the logged message.

### Environment info
- OS: Ubuntu Linux 16.04
- Python version: 2 or 3 
- Sphinx version: v1.7.4

--- a/CHANGES
+++ b/CHANGES
@@ -40,6 +40,7 @@ Bugs fixed
 * #4980: latex: Explicit labels on code blocks are duplicated
 * #4919: node.asdom() crashes if toctree has :numbered: option
 * #4914: autodoc: Parsing error when using dataclasses without default values
+* #5007: sphinx-build crashes when error log contains a "%" character
 
 Testing
 --------

--- a/sphinx/util/logging.py
+++ b/sphinx/util/logging.py
@@ -373,7 +373,7 @@ class WarningIsErrorFilter(logging.Filter):
             location = getattr(record, 'location', '')
             try:
                 message = record.msg % record.args
-            except TypeError:
+            except (TypeError, ValueError):
                 message = record.msg  # use record.msg itself
 
             if location:

